lib/nrf_inbuilt_key/nrf_inbuilt_key.c

```diff
@@ -0,0 +1,307 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <at_cmd.h>
+#include <at_cmd_parser/at_cmd_parser.h>
+#include <at_cmd_parser/at_params.h>
+#include <bsd_limits.h>
+#include <nrf_inbuilt_key.h>
+#include <stdio.h>
+#include <string.h>
+#include <zephyr.h>
+
+#define NRF_INBUILT_KEY_OP_LS "AT%CMNG=1"
+#define NRF_INBUILT_KEY_OP_RD "AT%CMNG=2"
+#define NRF_INBUILT_KEY_OP_WR "AT%CMNG=0"
+#define NRF_INBUILT_KEY_OP_RM "AT%CMNG=3"
+
+#define NRF_INBUILT_KEY_CHAR_SIZE_MAX_INT32 (10)
+#define NRF_INBUILT_KEY_CR_CHAR_SIZE (1)
+#define NRF_INBUILT_KEY_LF_CHAR_SIZE (1)
+#define NRF_INBUILT_KEY_QUOTES_CHAR_SIZE (1)
+#define NRF_INBUILT_KEY_ZERO_TERMINATOR_SIZE (1)
+#define NRF_INBUILT_KEY_COMMA_SIZE (1)
+#define NRF_INBUILT_KEY_INVALID_HANDLE (-1)
+
+#define VALIDATE_LENGTH(param, buffer_size)                                    \
+	do {                                                                   \
+		if (param > buffer_size) {                                     \
+			return -ENOBUFS;                                       \
```

Comment:
No.

Macro's that change the control flow are not OK.

Reply:
I just noticed that other modules where allowed. I can change it to do a full if-test in place.